libavcodec/jpeg2000dwt.c

```diff
@@ -530,7 +536,10 @@ static void dwt_decode97_int(DWTContext *s, int32_t *t)
     }
 
     for (i = 0; i < w * h; i++)
-        data[i] = (data[i] + ((1LL<<I_PRESHIFT)>>1)) >> I_PRESHIFT;
+        // Shifting down to the binary point.
+        // In FF_DWT97_INT, the binary point of the input coefficients is 1 bit above from the LSB.
+        // So, we need `>> (I_PRESHIFT + 1)`  here.
```

Comment:
```suggestion
        // We shift down by `I_PRESHIFT + 1` because the input coefficients `datap[]` were shifted down to 1 bit above from the binary point.
```